src/cljs/rems/util.cljs

```diff
@@ -134,18 +134,15 @@
   original string, except that all substrings that resemble a link have
   been changed to hiccup links."
   [s]
-  (when s
-    (let [splitted (-> s
-                       (str/replace link-regex #(str "\t" %1 "\t"))
-                       (str/split "\t"))
-          link? (fn [s] (re-matches link-regex s))
-          text-to-url (fn [s] (if (re-matches #"^(http://|https://).*" s)
-                                s
-                                (str "http://" s)))]
-      (map #(if (link? %)
-              [:a {:target :_blank :href (text-to-url %)} %]
-              %)
-           splitted))))
+  (for [substring (some-> s
+                          (str/replace link-regex #(str "\t" %1 "\t"))
+                          (str/split "\t"))
+        :let [url (when (re-matches link-regex substring)
+                    (if-not (str/starts-with? substring "http")
+                      (str "http://" substring) substring))]]
+    (if url
+      ^{:key (random-uuid)} [:a {:target :_blank :href url} substring]
+      substring)))
```

Comment:
So what changed here? I can see just adding `:key` but that could be avoided in the caller?

Reply:
how would you avoid `:key` in the caller side without sacrificing readability? the common use case looks like this
```clojure
(linkify raw-title)
```

Reply:
Something like this? `(into [:span] (linkify raw-title))`

Reply:
I feel like we should perhaps improve the contract of the function to return hiccup. Using random UUIDs feels like a smell.

Reply:
maybe we could do the `(into)` dance in the utility function itself then?

Reply:
looking at `rems.test-util/test-linkify` the original usage seems to have been more akin to string than hiccup:
```clojure
(testing "retain original string"
      (is (= (apply str (linkify "a b c") "a b c"))))
```

Reply:
We are free to fix it :+1:. Creating links in the middle of a string doesn't work so it should be a kind of a component and every place of use could just be fixed one way or another. But how about we create a new issue out of it?

Reply:
i reverted the change. feel free to create the issue

Reply:
:+1: Created https://github.com/CSCfi/rems/issues/2899